### DISCUSSION TITLE
I was stuck in the "Get Authorization Code" step

### DISCUSSION THREAD

**Discussion by Aceak** [spencerwooo/onedrive-vercel-index#502] [Q&A] [answered]
I was stuck in the "Get Authorization Code" step and failed to extract the authorization code
![472309f790529822c05570a0c7ca7bcb0a46d46b](https://user-images.githubusercontent.com/49723740/155727258-baefac81-b2de-4595-b602-d7c4b12f9a25.jpg)
I've changed accounts several times, all stuck at this step, and I can make sure that each step is correct

**Comment by zkeq** [comment]
Please make sure your links start with `http://`

**Comment by zkeq** [comment, marked as answer]
maybe #319  #310 ?

**Reply by Aceak** [reply]
I've tried both  ` http:// ` and ` https:// ` , but they didn't work , and the redirected link started ` https:// `

**Reply by zkeq** [reply]
Can you provide the first half of this link ?

**Reply by zkeq** [reply]
> I've tried both `http://` and `https://` , but they didn't work , and the redirected link started `https://`

Your link should be something like this to be recognized, maybe you should check something is wrong?

```
http://localhost/?code=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

**Reply by Aceak** [reply]
It prompts that I used an invalid Client_id
`` http://localhost/?error=invalid_client&error_description=AADSTS650051%3a+Using+application+%27OneCloud%27+is+currently+not+supported+for+your+organization+std.uestc.edu.cn+because+it+is+in+an+unmanaged+state.+An+administrator+needs+to+claim+ownership+of+the+company+by+DNS+validation+of+std.uestc.edu.cn+before+the+application+OneCloud+can+be+provisioned.%0d%0aTrace+ID%3a+263ff761-ec7a-48f6-8fa9-089219a75000%0d%0aCorrelation+ID%3a+63c90012-53b5-4c5e-95e5-37e53e118578%0d%0aTimestamp%3a+2022-02-26+13%3a03%3a09Z# ``

**Reply by zkeq** [reply]
> It prompts that I used an invalid Client_id `http://localhost/?error=invalid_client&error_description=AADSTS650051%3a+Using+application+%27OneCloud%27+is+currently+not+supported+for+your+organization+std.uestc.edu.cn+because+it+is+in+an+unmanaged+state.+An+administrator+needs+to+claim+ownership+of+the+company+by+DNS+validation+of+std.uestc.edu.cn+before+the+application+OneCloud+can+be+provisioned.%0d%0aTrace+ID%3a+263ff761-ec7a-48f6-8fa9-089219a75000%0d%0aCorrelation+ID%3a+63c90012-53b5-4c5e-95e5-37e53e118578%0d%0aTimestamp%3a+2022-02-26+13%3a03%3a09Z#`

Maybe you should try this step : https://ovi.swo.moe/docs/advanced#using-your-own-clientid-and-clientsecret-1

**Reply by Aceak** [reply]
The redirect link started with https because I used an anti-proxy software, and then I turned off the anti-proxy, but it still failed

**Reply by Aceak** [reply]
> > It prompts that I used an invalid Client_id `http://localhost/?error=invalid_client&error_description=AADSTS650051%3a+Using+application+%27OneCloud%27+is+currently+not+supported+for+your+organization+std.uestc.edu.cn+because+it+is+in+an+unmanaged+state.+An+administrator+needs+to+claim+ownership+of+the+company+by+DNS+validation+of+std.uestc.edu.cn+before+the+application+OneCloud+can+be+provisioned.%0d%0aTrace+ID%3a+263ff761-ec7a-48f6-8fa9-089219a75000%0d%0aCorrelation+ID%3a+63c90012-53b5-4c5e-95e5-37e53e118578%0d%0aTimestamp%3a+2022-02-26+13%3a03%3a09Z#`
> 
> Maybe you should try this step : https://ovi.swo.moe/docs/advanced#using-your-own-clientid-and-clientsecret-1

I've used this method, and I've even tried changing accounts

**Reply by zkeq** [reply]
> The redirect link started with https because I used an anti-proxy software, and then I turned off the anti-proxy, but it still failed

Your link doesn't seem to be in the beginning with this.

```
http://localhost/?code=
```
Instead of this one
```
http://localhost/?error=
```
I think this is the problem..

Try your own `clientId` and `clientSecret` may solve the problem: https://ovi.swo.moe/docs/advanced#using-your-own-clientid-and-clientsecret-1

**Reply by zkeq** [reply]
> I've used this method, and I've even tried changing accounts

Can I have a look at your code warehouse?I think according to the document for the steps should be normal use?

**Reply by Aceak** [reply]
I think i can try default `clienid` ,  I used the default `clientid` and failed at the time, but I didn't pay attention to the redirect link at the time

**Reply by Aceak** [reply]
> > I've used this method, and I've even tried changing accounts
> 
> Can I have a look at your code warehouse?I think according to the document for the steps should be normal use?

It is in the `OneCloud` repository , I had forked the sourcecode ,so `OneCloud` repository is Public

**Reply by zkeq** [reply]
> > > I've used this method, and I've even tried changing accounts
> > 
> > 
> > Can I have a look at your code warehouse?I think according to the document for the steps should be normal use?
> 
> It is in the `OneCloud` repository , I had forked the sourcecode ,so `OneCloud` repository is Public

I cloned your code and deployed it to one of my branches, tested my own `E5` account and it seems to be working..

![1](https://user-images.githubusercontent.com/62864752/155845006-912f8140-9de0-449a-ae07-2c3e0347a947.png)

This is the deployed address:
```
https://onedrive-vercel-china-git-test-zkeq.vercel.app/onedrive-vercel-index-oauth/step-2/
```

Here is part of the code I got

```
http://localhost/?code=0.AVUAFvKegAkBzUqgHx4OkxeIRgZbklmkfCBJu6gYP3q5HR2IAPg.AQABAAIAAAD--DLA3VO7Qrddg
```

**Reply by Aceak** [reply]
It may be a problem with my `A1` account permissions, I will try again with an `E5` account
Thanks a lot

**Reply by spencerwooo** [reply]
Oh, that means your edu account requires admin authentication, which you probably won't have. You would need to use your own client id and secrets. Check: https://ovi.swo.moe/docs/advanced#using-your-own-clientid-and-clientsecret

**Comment by spencerwooo** [comment]
Yes, and can you share a redacted redirect URL here? (meaning that you remove some sensitive parts of the URL)

**Comment by qcmuu** [comment]
I followed the instruction and got the same issues. 

http://localhost/?error=invalid_client&error_description=AADSTS650051: Using application 'mail_test' is currently not supported for your organization xxx.edu because it is in an unmanaged state. An administrator needs to claim ownership of the company by DNS validation of stu.edu.cn before the application mail_test can be provisioned.

**Comment by KuroProjects** [comment]
![image](https://user-images.githubusercontent.com/92580272/157070233-a5c0eb5b-fb11-4a21-9787-17b55eb15691.png)
I followed the instructions and i am getting this kind of issue
where i am doing wrong any help?

**Reply by Aceak** [reply]
Did you used education acount? An administrator needs to claim ownership of the company by DNS validation of stu.edu.cn before the application mail_test can be provisioned.
